packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -364,38 +391,63 @@ export class PartialSequenceLengths {
                 penultPartialLen = pLen;
             }
         }
+        const len = penultPartialLen !== undefined ? penultPartialLen.len + seqSeglen : seqSeglen;
         if (seqPartialLen === undefined) {
-            // len will be assigned below, making this assertion true.
-            // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
             seqPartialLen = {
                 clientId,
+                len,
                 seglen: seqSeglen,
                 seq,
-            } as PartialSequenceLength;
+            };
             partialLengths.push(seqPartialLen);
         } else {
             seqPartialLen.seglen = seqSeglen;
+            seqPartialLen.len = len;
             // Assert client id matches
         }
-        if (penultPartialLen !== undefined) {
-            seqPartialLen.len = seqPartialLen.seglen + penultPartialLen.len;
-        } else {
-            seqPartialLen.len = seqPartialLen.seglen;
-        }
     }
-    public minLength = 0;
-    public segmentCount = 0;
-    public partialLengths: PartialSequenceLength[] = [];
-    public clientSeqNumbers: PartialSequenceLength[][] = [];
 
-    constructor(public minSeq: number) {
-    }
+    /**
+     * Length of the block this PartialSequenceLength corresponds to when viewed at `minSeq`.
+     */
+    private minLength = 0;
+
+    /**
+     * Total number of segments in the subtree rooted at the block this PartialSequenceLength corresponds to.
+     */
+    private segmentCount = 0;
+
+    /**
+     * List of PartialSequenceLength objects--ordered by increasing seq--giving length information about
+     * the block associated with this PartialSequenceLengths object.
+     *
+     * `partialLengths[i].len` contains the length of this block considering only sequenced segments with
+     * `sequenceNumber <= partialLengths[i].seq`.
+     */
+    private readonly partialLengths: PartialSequenceLength[] = [];
+
+    /**
+     * clientSeqNumbers[clientId] is a list of partial lengths for sequenced ops which either:
+     * - were submitted by `clientId`.
+     * - deleted a range containing segments that were concurrently deleted by `clientId`
+     *
+     * The second case is referred to as the "overlapping delete" case. It is necessary to avoid double-counting
+     * the removal of those segments in queries including clientId.
+     */
+    private readonly clientSeqNumbers: PartialSequenceLength[][] = [];
```

Comment:
Unrelated to the actual change in the PR, but could this benefit from being a map instead of an array? Are clientIds always small integers? Thinking of some case where we get a clientId = 10,000 or something and have to keep a big sparse array in memory.

Reply:
Yeah, these are always small integers. See logic in `Client.getOrAddShortId`.